---
title: "Striatum RNAseq Differential Expression"
output:
  pdf_document: default
  html_notebook: default
---

--------
***Outlier STN7 can be excluded due to poor RNA quality (low 260/230)***
***Outliers excluded due to low 260/280 < 1.90 and/or 260/230 < 1.50   -- STN5 , STN7 were not sequenced by novogene***
-----------


#Load libraries
```{r include=FALSE}
library(ggplot2)
library(tidyr)
library(tidyverse)
library(dplyr)
library(BiocManager)
library(DESeq2)
library(pheatmap)
library(AnnotationDbi)
library(Rattus.norvegicus)
library(EnhancedVolcano)
library(clusterProfiler)
library(fgsea)
library(pathview)
library(enrichplot)
library(viridis)
library(babelgene)
library(DOSE)
library(org.Hs.eg.db)
library(patchwork)
```

#***Using DeSeq2***
We can now construct a DESeqDataSet object to formulate a starting point for our analysis. You need to add an appropriate design for analysis

#removing outliers identified to clean and rerun analysis
```{r}
se1<- se[,se$tissue_type == "striatum"] ##change tissue type here
se1<-se1[ , !se1$animal_ID ==c("7-347","5-115","19-067")] #remove specific outliers here
F_se<-se1[,se1$sex == "F"]
M_se<-se1[,se1$sex == "M"]
test<-M_se #[ , !se1$animal_ID ==] #remove specific outliers here
colData(test)
#test$group<-paste(test$exposure, test$sex, sep = "_") #create a group variable for easy comparison
#test$group

#test$group<-factor(test$group, levels = c("Sham_M","Exp_M"))
dds <- DESeqDataSet(test, 
                    design = ~ exposure) #comparison done between high treatment and vehicle?
mdds<-dds
```

#***Exploratory analysis and visualization***
*transformations*
The variance stabilizing transformation (VST) a goal of stablizing the variance across the range of values. produce log2-like values for high counts. 
```{r}
colData(dds)

vsd <- vst(dds)
```

Now for visuals, ploat a principal components analysis (PCA) using  ggplot2
```{r}
data <- plotPCA(vsd, 
                  intgroup = 'exposure' , 
                  returnData=TRUE)
data
percentVar <- round(100 * attr(data, 
                                 "percentVar"))
```
#PCA Plot
```{r}
tiff(file = 'Striatum/PCA_plots/Male_CRBLM_PCA.tiff', units="in", width=8, height=5, res=1000)
PCA <-ggplot(data, aes(PC1, 
                   PC2, 
                   color=exposure,
                   shape= exposure)) + 
  stat_ellipse(aes(color= exposure))+
  ggtitle("Male Striatum PCA")+
  geom_point(size=3) +
  scale_color_manual(values = c("#999900", "#FF9933")) +
  scale_shape_manual(values = c(19, 15)) +
  xlab(paste0("PC1: ",
              percentVar[1],
              "% variance")) +
  ylab(paste0("PC2: ",
              percentVar[2],
              "% variance"))+ 
  theme_bw() + 
  theme(panel.border =element_rect(colour = "black", fill=NA, size=1), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black"))
PCA
dev.off()
PCA
```

#***Differential Expression***
Lets run a differential expression pipeline using the raw counts

This function will print out a message for the various steps it performs. These are described in more detail in the manual page for DESeq, which can be accessed by typing ?DESeq. Briefly these are: the estimation of size factors (controlling for differences in the sequencing depth of the samples), the estimation of dispersion values for each gene, and fitting a generalized linear model.
```{r}
dds <- DESeq(dds)
resultsNames(dds)
```
Calling results without any arguments will extract the estimated log2 fold changes and p values for the last variable in the design formula. If there are more than 2 levels for this variable, results will extract the results table for a comparison of the last level over the first level. Treatment 6 vs vehicle
```{r}
res<-results(dds)
M_res <- results(dds)
summary(M_res)

sum(M_res$log2FoldChange > 1 & M_res$pvalue < 0.05, na.rm=TRUE)
```

#***Annotating results***
```{r}
columns(Rattus.norvegicus)
```

How to assign actual gene names to our counts. Using an annotation package for Mus musculus. 
```{r}
M_res$symbol <- mapIds(Rattus.norvegicus, #which annotation package model to use
                     keys=row.names(M_res), #using rownames
                     column="SYMBOL", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
```

```{r}
M_res$TXname <- mapIds(Rattus.norvegicus, #which annotation package model to use
                     keys=row.names(M_res), #using rownames
                     column="TXNAME", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
```

```{r}
M_res$ONTOLOGY <- mapIds(Rattus.norvegicus,
                     keys=row.names(M_res),
                     column="ONTOLOGYALL",
                     keytype="GENEID",
                     multiVals="first")
head(M_res)
```

```{r}
M_res$Genename <- mapIds(Rattus.norvegicus,
                     keys=row.names(M_res),
                     column="GENENAME", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
```

```{r}
M_res$GOID <- mapIds(Rattus.norvegicus,
                     keys=row.names(M_res),
                     column="GO", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
```

```{r}
M_res$ensembl <- mapIds(Rattus.norvegicus,
                     keys=row.names(M_res),
                     column="ENSEMBL", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
```
```{r}
M_res$ENTREZID <- mapIds(Rattus.norvegicus,
                     keys=row.names(M_res),
                     column="ENTREZID", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
```

```{r}
M_resOrdered <- M_res[order(M_res$padj),] #reorder the genes based on significance
head(M_resOrdered, 10)
```

#Plotting results
###volcano plots
```{r}
tiff(file = 'Striatum/volcano_plots/M_volcano_plot_stn.tiff', units="in", width=8, height=5, res=1000)
M_vol1<- EnhancedVolcano(M_res,
    lab = M_res$symbol,
    x = 'log2FoldChange',
    y = 'pvalue',
    title = "Male",
    subtitle = "Striatum",
    ylim = c(0, -log10(10e-8)),
    xlim = c(log2(10e-3), -log2(10e-3)),
    FCcutoff = 1.0,
    pCutoff = 0.05,
    labSize = 3.0,
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 3.0,
    gridlines.major = FALSE,
    gridlines.minor = FALSE,
    border = 'full',
    borderWidth = 0.5,
    borderColour = 'black')
M_vol1
dev.off()
M_vol1
```
#Export Male TSV results
```{r}
M_resOrdered <- as.data.frame(M_resOrdered)
write_tsv(M_resOrdered, file="Striatum/results/Male_STN_results.tsv")
write.csv(M_resOrdered, file="Striatum/results/Male_STN_results.csv")
write.csv(as.data.frame(assay(M_se)), file= "Striatum/results/Male_STN_counts.csv")
```

##Female Results (comparison between sham and exposed females)
```{r}
test<-F_se #[ , !se1$animal_ID ==] #remove specific outliers here
colData(test)
#test$group<-paste(test$exposure, test$sex, sep = "_") #create a group variable for easy comparison
#test$group

#test$group<-factor(test$group, levels = c("Sham_M","Exp_M"))
dds <- DESeqDataSet(test, 
                    design = ~ exposure) #comparison done between high treatment and vehicle?
fdds<-dds
```

#***Exploratory analysis and visualization***
*transformations*
The variance stabilizing transformation (VST) a goal of stablizing the variance across the range of values. produce log2-like values for high counts. 
```{r}
colData(dds)

vsd <- vst(dds)
```

Now for visuals, ploat a principal components analysis (PCA) using  ggplot2
```{r}
data <- plotPCA(vsd, 
                  intgroup = 'exposure' , 
                  returnData=TRUE)
data
percentVar <- round(100 * attr(data, 
                                 "percentVar"))
```
#PCA Plot
```{r}
tiff(file = 'Striatum/PCA_plots/Female_STN_PCA.tiff', units="in", width=8, height=5, res=1000)
PCA <-ggplot(data, aes(PC1, 
                   PC2, 
                   color=exposure,
                   shape= exposure)) + 
  stat_ellipse(aes(color= exposure))+
  ggtitle("Female Striatum PCA")+
  geom_point(size=3) +
  scale_color_manual(values = c("#3399FF", "#9933FF")) +
  scale_shape_manual(values = c(17, 18)) +
  xlab(paste0("PC1: ",
              percentVar[1],
              "% variance")) +
  ylab(paste0("PC2: ",
              percentVar[2],
              "% variance"))+ 
  theme_bw() + 
  theme(panel.border = element_rect(colour = "black", fill=NA, size=1), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black"))
PCA
dev.off()
PCA
```
#***Differential Expression***
Lets run a differential expression pipeline using the raw counts

This function will print out a message for the various steps it performs. These are described in more detail in the manual page for DESeq, which can be accessed by typing ?DESeq. Briefly these are: the estimation of size factors (controlling for differences in the sequencing depth of the samples), the estimation of dispersion values for each gene, and fitting a generalized linear model.
```{r}
dds <- DESeq(dds)
resultsNames(dds)
```
Calling results without any arguments will extract the estimated log2 fold changes and p values for the last variable in the design formula. If there are more than 2 levels for this variable, results will extract the results table for a comparison of the last level over the first level. Treatment 6 vs vehicle
```{r}
F_res <- results(dds)
F_res
summary(F_res)
sum(F_res$log2FoldChange < -1 & F_res$pvalue < 0.05, na.rm=TRUE)
```

#***Annotating results***
How to assign actual gene names to our counts. Using an annotation package for Mus musculus. 
```{r}
F_res$symbol <- mapIds(Rattus.norvegicus, #which annotation package model to use
                     keys=row.names(F_res), #using rownames
                     column="SYMBOL", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
```

```{r}
F_res$TXname <- mapIds(Rattus.norvegicus, #which annotation package model to use
                     keys=row.names(F_res), #using rownames
                     column="TXNAME", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")

```

```{r}
F_res$ONTOLOGY <- mapIds(Rattus.norvegicus,
                     keys=row.names(F_res),
                     column="ONTOLOGYALL",
                     keytype="GENEID",
                     multiVals="first")

```

```{r}
F_res$Genename <- mapIds(Rattus.norvegicus,
                     keys=row.names(F_res),
                     column="GENENAME", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")

```

```{r}
F_res$GOID <- mapIds(Rattus.norvegicus,
                     keys=row.names(F_res),
                     column="GO", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")

```

```{r}
F_res$ensembl <- mapIds(Rattus.norvegicus,
                     keys=row.names(F_res),
                     column="ENSEMBL", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")

```

```{r}
F_res$ENTREZID <- mapIds(Rattus.norvegicus,
                     keys=row.names(F_res),
                     column="ENTREZID", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")

```

```{r}
F_resOrdered <- F_res[order(F_res$padj),] #reorder the genes based on significance
head(F_resOrdered, 10)
```

#***Female volcano plot***
```{r}
tiff(file = 'Striatum/volcano_plots/F_volcano_plot_STN.tiff', units="in", width=8, height=5, res=1000)
F_vol1<- EnhancedVolcano(F_res,
    lab = F_res$symbol,
    x = 'log2FoldChange',
    y = 'pvalue',
    title = "Female",
    subtitle = "Striatum",
    ylim = c(0, -log10(10e-8)),
    xlim = c(log2(10e-3), -log2(10e-3)),
    FCcutoff = 1.0,
    pCutoff = 0.05,
    labSize = 3.0,
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 3.0,
    gridlines.major = FALSE,
    gridlines.minor = FALSE,
    border = 'full',
    borderWidth = 0.5,
    borderColour = 'black')
F_vol1
dev.off()
F_vol1
```

#***Exporting_female_report***
```{r}
F_resOrdered <- as.data.frame(F_resOrdered)
write_tsv(F_resOrdered, file="Striatum/results/Female_STN_results.tsv")
write.csv(F_resOrdered, file="Striatum/results/Female_STN_results.csv")
write.csv(as.data.frame(assay(F_se)), file= "Striatum/results/Female_STN_counts.csv")
```


----------------
#Gene enrichment analysis (GSEA)**
```{r}
x<- M_res[order(M_res$log2FoldChange, decreasing = TRUE),] #reorder the genes based on significance
head(x)
x<-na.omit(x)
gene_list<-x$log2FoldChange
names(gene_list)<-x$ensembl
head(gene_list)
```

```{r}
x<- F_res[order(F_res$log2FoldChange, 
                decreasing = TRUE),] #reorder the genes based on significance
x<-na.omit(x)
gene_list<-x$log2FoldChange
names(gene_list)<-x$ensembl
head(gene_list)
```

#Compare Cluster Gene, placing all treatment groups in the same dotplot
```{r}
x<- M_res[order(M_res$log2FoldChange, decreasing = TRUE),]  #reorder the genes based on significance
x<-na.omit(x)

gene_list<-x$log2FoldChange
g_list<-data.frame(gene_list)
colnames(g_list)<-"LFC"

g_list$entrez<-x$ENTREZID


hum_orthos<- orthologs(genes = g_list$entrez, species = "rat", human = FALSE)

DO_data<-merge(g_list, hum_orthos, by.x = "entrez", by.y= "entrez", all.x = TRUE)

#rename the listed Fold changes for the genes with the human equivalent for DO
genes1<-DO_data$LFC
names(genes1)<-DO_data$human_entrez
genes1<-genes1[order(genes1, decreasing=TRUE)]
head(genes1,10)
```

```{r}
x<- F_res[order(F_res$log2FoldChange, decreasing = TRUE),]  #reorder the genes based on significance
x<-na.omit(x)

gene_list<-x$log2FoldChange
g_list<-data.frame(gene_list)
colnames(g_list)<-"LFC"

g_list$entrez<-x$ENTREZID


hum_orthos<- orthologs(genes = g_list$entrez, species = "rat", human = FALSE)

DO_data<-merge(g_list, hum_orthos, by.x = "entrez", by.y= "entrez", all.x = TRUE)

#rename the listed Fold changes for the genes with the human equivalent for DO
genes2<-DO_data$LFC
names(genes2)<-DO_data$human_entrez
genes2<-genes2[order(genes2, decreasing=TRUE)]
head(genes2,10)
```

##*STN*
###Gene
```{r}
inputList <- list(`Male` = genes1, `Female` = genes2)
STN_groups <- compareCluster(geneClusters=inputList,  fun = "gseGO", OrgDb=org.Hs.eg.db, pvalueCutoff = 0.05, pAdjustMethod = "none")


data<-data.frame(STN_groups)
write.csv(data ,file="Striatum/Enrichment/Grouped_STN_Gene_enrichment.csv")
```


```{r}
tiff(file = 'Striatum/Enrichment/STN_grouped_gene_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot1<-dotplot(STN_groups,showCategory=3, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 45)))+
  theme(axis.text.y = element_text(size = 12), 
        strip.background = element_blank(),
        strip.text.x = element_text(size = 12),
        axis.text.x = element_text(size = 14),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 10),
        legend.key.width= unit(0.3, 'cm'))  
dot1
dev.off()
dot1
```

###KEGG
```{r}
STN_groups <- compareCluster(geneClusters=inputList,  fun = "gseKEGG", pvalueCutoff = 0.05, pAdjustMethod = "none")

data<-data.frame(STN_groups)
write.csv(data ,file="Striatum/Enrichment/Grouped_STN_KEGG_enrichment.csv")
```


```{r}
tiff(file = 'Striatum/Enrichment/STN_grouped_kegg_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot2<-dotplot(STN_groups,showCategory=3, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 45)))+
  theme(axis.text.y = element_text(size = 13), 
        strip.background = element_blank(),
        strip.text.x = element_text(size = 14),
        axis.text.x = element_text(size = 12),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 10),
        legend.key.width= unit(0.3, 'cm'))  
dot2
dev.off()
dot2
```


###Disease Ontology
```{r}
STN_groups_DO <- compareCluster(geneClusters=inputList,  fun = "gseDO", pvalueCutoff = 0.05, pAdjustMethod = "none")
data<-data.frame(STN_groups)
write.csv(data ,file="Striatum/Enrichment/Grouped_STN_Disease_enrichment.csv")
```


```{r}
tiff(file = 'Striatum/Enrichment/STN_grouped_disease_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot3<-dotplot(STN_groups_DO,showCategory=3, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 45)))+
  theme(axis.text.y = element_text(size = 12), 
        strip.background = element_blank(),
        strip.text.x = element_text(size = 14),
        axis.text.x = element_text(size = 12),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 10),
        legend.key.width= unit(0.3, 'cm'))  
dot3
dev.off()
dot3
```

```{r}
STN_big<- dot1/dot2/dot3

tiff(file = 'Striatum/Enrichment/STN_dot_plots_big.tiff', units="in",width=10, height=15, res=500)
STN_big + plot_annotation(tag_levels = 'A') & 
  theme(plot.tag = element_text(size = 16))
dev.off()

```

```{r}
# Produce the native KEGG plot (PNG)
#female_Striatum <- pathview(gene.data=gene_list, pathway.id="04360", species = kegg_organism)
```
